### PR TITLE
[#2222] Zoom slider for superscrollbar WIP. Feeback is welcome.

### PR DESCRIPTION
**Pull request by ScottDowne**

### CODE REVIEW

File: src/timeline/super-scrollbar.js [hunk 1]
```diff
@@ -190,138 +190,94 @@ define( [ "util/lang", "text!layouts/super-scrollbar.html" ],
...
+          leftPosition;
+
+      // make sure our growth is at least a pixel on either side.
+      if ( ( pixelGrowth > -1 && pixelGrowth < 1 ) ) {
+        return;
+      }
 
-      leftPosition = ( leftPosition < 0 ? 0 : leftPosition ) / _element.clientWidth;
-      rightPosition = ( rightPosition < 0 ? 0 : rightPosition) / _element.clientWidth;
+      rightPosition = ( 1 - ( ( viewLeft + viewWidth ) / rectWidth ) ) + halfScale,
```
Comment (mjschranz) [inline src/timeline/super-scrollbar.js]: semi colon at the end.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/4fd7057ccab7755/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/036d6766f890f02/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/4fd7057ccab7755/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/036d6766f890f02/output.txt

Total script time: 3.00 mins

File: css/super-scrollbar.less [hunk 2]
```diff
@@ -1,50 +1,83 @@
...
+    > .tick:nth-child( 6 ) {
+        margin-left: 32px;
     }
   }
 }
 
+
+#butter-super-scrollbar-inner-container {
+  background: darken( @baseLight, 3% );
+  display: block;
```
Comment (secretrobotron) [inline css/super-scrollbar.less]: This should already be `display: block`, right?

File: css/super-scrollbar.less [hunk 3]
```diff
@@ -1,50 +1,83 @@
...
+    background: #EEE;
+    border: 1px solid #CCC;
+    width: 100px;
+    height: 6px;
+    border-radius: 6px;
+    left: 12px;
+    top: 13px;
+    z-index: 2;
+
+    > .tick {
```
Comment (secretrobotron) [inline css/super-scrollbar.less:38]: Spacing (indentation) on all of these tick children.

File: src/timeline/super-scrollbar.js [hunk 4]
```diff
@@ -19,32 +19,31 @@ define( [ "util/lang", "text!layouts/super-scrollbar.html" ],
...
-                                  // buttons are clicked. Keeping this value near 1 is a good idea since
-                                  // width = width * ZOOM_EXPAND_AMOUNT.
-
-      TRANSITION_TIMEOUT = 100,   // Length of timeout for viewport-transition to stay alive.
-      ZOOM_PULSE_DURATION = 250;  // Length of time between zoom pulses.
+      ARROW_MIN_WIDTH_CLASS = "super-scrollbar-small";
 
   return function( outerElement, innerElement, boundsChangedCallback, media ) {
-    var _element = LangUtils.domFragment( SUPER_SCROLLBAR_LAYOUT ).querySelector( "#butter-super-scrollbar-container" ),
+    var _outer = LangUtils.domFragment( SUPER_SCROLLBAR_LAYOUT ).querySelector( "#butter-super-scrollbar-outer-container" ),
```
Comment (secretrobotron) [inline src/timeline/super-scrollbar.js]: You can probably use the form `LangUtils.domFragment( SUPER_SCROLLBAR_LAYOUT, "#butter-super-scrollbar-outer-container" )`

File: src/timeline/super-scrollbar.js [hunk 5]
```diff
@@ -181,136 +180,95 @@ define( [ "util/lang", "text!layouts/super-scrollbar.html" ],
...
-     * as percentage values (0 - 1). This action has the consequence of zooming out the
-     * track container viewport, since more viewable area is desired.
+     * Scales the viewport by a percentage value (0 - 1). The viewport grows or shrinks
+     * to cover less or more area, and calls _boundsChangedCallback with the new (left, width) combination
+     * as percentage values (0 - 1). This action has the consequence of zooming the
+     * track container viewport in or out.
      *
      * A left and right position are calculated by moving them a set amount from their current
-     * positions away from the mid-point of the viewport. A new width value is also calculated
+     * positions toward the mid-point of the viewport. A new width value is also calculated
```
Comment (secretrobotron) [inline src/timeline/super-scrollbar.js]: You can't use `toward` here anymore. Maybe use `around`.

File: src/timeline/super-scrollbar.js [hunk 6]
```diff
@@ -181,136 +180,95 @@ define( [ "util/lang", "text!layouts/super-scrollbar.html" ],
...
+      window.removeEventListener( "mouseup", zoomSliderMouseDown, false );
+      window.removeEventListener( "mousemove", zoomSliderMouseMove, false );
+      _zoomSliderContainer.addEventListener( "mousedown", zoomSliderMouseDown, false );
+    }
 
-      leftPosition = ( leftPosition < 0 ? 0 : leftPosition ) / _element.clientWidth;
-      rightPosition = ( rightPosition < 0 ? 0 : rightPosition) / _element.clientWidth;
+    function zoomSliderMouseMove( e ) {
+      e.preventDefault();
+      e.stopPropagation();
```
Comment (secretrobotron) [inline src/timeline/super-scrollbar.js]: Why is this here? Are you trying to stop things from listening to when the slider moves?

Comment (secretrobotron) [inline src/timeline/super-scrollbar.js]: There are a few of these you can probably get rid of.

File: src/timeline/super-scrollbar.js [hunk 7]
```diff
@@ -181,136 +180,95 @@ define( [ "util/lang", "text!layouts/super-scrollbar.html" ],
...
-    function zoomIn() {
-      var halfWidth = ( _viewPort.clientWidth * ZOOM_EXPAND_AMOUNT - _viewPort.clientWidth ) / 2,
-          viewportRect = _viewPort.getBoundingClientRect(),
-          leftPosition = _viewPort.offsetLeft + halfWidth,
-          rightPosition = _rect.right - viewportRect.right + halfWidth,
-          newWidth = viewportRect.width - halfWidth * 2;
+    function zoomSliderMouseUp( e ) {
+      e.stopPropagation();
+      _viewPort.classList.remove( "viewport-transition" );
+      window.removeEventListener( "mouseup", zoomSliderMouseDown, false );
```
Comment (secretrobotron) [inline src/timeline/super-scrollbar.js]: This looks wrong.
